content/program-security.md

```diff
@@ -0,0 +1,741 @@
+) -> ProgramResult {    
+    msg!("Updating movie review...");
+
+    // Get Account iterator
+    let account_info_iter = &mut accounts.iter();
+
+    // Get accounts
+    let initializer = next_account_info(account_info_iter)?;
+    let pda_account = next_account_info(account_info_iter)?;
+
+    msg!("unpacking state account");
+    let mut account_data = try_from_slice_unchecked::<MovieAccountState>(&pda_account.data.borrow()).unwrap();
+    msg!("revie title: {}", account_data.title);
+
+    msg!("checking if movie account is initialized");
+    if !account_data.is_initialized() {
+        msg!("Account is not initialized");
+        return Err(ReviewError::UnitializedAccount.into());
+    }
+
+    if pda_account.owner != program_id {
+      return Err(ProgramError::InvalidOwner)
+    }
+
+    // Derive PDA and check that it matches client
+    let (pda, _bump_seed) = Pubkey::find_program_address(&[initializer.key.as_ref(), account_data.title.as_bytes().as_ref(),], program_id);
+
+    if pda != *pda_account.key {
+        msg!("Invalid seeds for PDA");
+        return Err(ReviewError::InvalidPDA.into())
+    }
+
+    if !initializer.is_signer {
+        msg!("Missing required signature");
+        return Err(ProgramError::MissingRequiredSignature)
+    }
+
+    let update_len: usize = 1 + 1 + (4 + description.len());
+    if update_len > 1000 - account_data.title.len() {
+        msg!("Data length is larger than 1000 bytes");
+        return Err(ReviewError::InvalidDataLength.into())
+    }
+
+    if rating > 5 {
+        msg!("Rating cannot be higher than 5");
+        return Err(ReviewError::RatingTooHigh.into())
+    }
+
+    msg!("Review before update:");
+    msg!("Title: {}", account_data.title);
+    msg!("Rating: {}", account_data.rating);
+    msg!("Description: {}", account_data.description);
+
+    account_data.rating = rating;
+    account_data.description = description;
+
+    msg!("Review after update:");
+    msg!("Title: {}", account_data.title);
+    msg!("Rating: {}", account_data.rating);
+    msg!("Description: {}", account_data.description);
+
+    msg!("serializing account");
+    account_data.serialize(&mut &mut pda_account.data.borrow_mut()[..])?;
+    msg!("state account serialized");
+
+    Ok(())
+}
+```
+
+Now, you can simply build and upgrade your program just as before. Clone [this branch](https://github.com/ixmorrow/movie-review-testing-script/tree/update-review-script)  of the testing script repo to test out your program’s new functionality. All you need to do is run the main function in the script and it will generate a keypair, create a review account, and update that review account in two separate transactions. Follow the link each transaction logs to the console to see all of the program logs in the block explorer. Play around with the script a little to change the movies/reviews it’s making and maybe even refactor it so it uses your local cli wallet to create them instead of a random keypair!
```

Comment:
@jamesrp13 do we want to use uppercase acronyms for PDA?

Project-wide, outside of this lesson, we have no uses of lowercase "pda" within a (non code) sentence, and ~20 uses capitalized.  Lemme know and I'll sort them out when yr wrapping up